### PR TITLE
Use explicitly sized integer types for Postgres-specific functions

### PR DESCRIPTION
**Pull request by kmicklas**
This is the first step of #496.

Relevant documentation:

 * `UNNEST ... WITH ORDINALITY`: https://www.postgresql.org/docs/12/queries-table-expressions.html
 * JSON functions: https://www.postgresql.org/docs/current/functions-json.html
 * pgcrypto: https://www.postgresql.org/docs/current/pgcrypto.html

### CODE REVIEW

File: beam-postgres/Database/Beam/Postgres/PgSpecific.hs [hunk 1]
```diff
@@ -1409,14 +1410,14 @@ pgUnnestArray (QExpr q) =
...
-data PgUnnestArrayWithOrdinalityTbl a f = PgUnnestArrayWithOrdinalityTbl (C f Int) (C f a)
+data PgUnnestArrayWithOrdinalityTbl a f = PgUnnestArrayWithOrdinalityTbl (C f Int64) (C f a)
   deriving Generic
 instance Beamable (PgUnnestArrayWithOrdinalityTbl a)
 
 -- | Introduce each element of the array as a row, along with the
 -- element's index
 pgUnnestArrayWithOrdinality :: QExpr Postgres s (V.Vector a)
-                            -> Q Postgres db s (QExpr Postgres s Int, QExpr Postgres s a)
+                            -> Q Postgres db s (QExpr Postgres s Int64, QExpr Postgres s a)
```
Comment (3noch) [inline beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: How are deciding which Int type to use?

Comment (kmicklas) [reply beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: Postgres documentation: https://www.postgresql.org/docs/12/queries-table-expressions.html

Comment (3noch) [reply beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: Oh I see it's the `bigint` here in the docs for `WITH ORDINALITY`... I wish there were a more obvious way to connect this choice with the docs.

Comment (3noch) [reply beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: I'm normally not a big fan of type aliases but perhaps it makes sense to define `type PgBigInt = Int64` to make it more obvious. But even then we have to go digging through docs again to see where the bigint shows up.

Comment (kmicklas) [reply beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: Hmm, I think I'd rather not use type aliases. Besides all the usual reasons, it seems a bit more confusing to have something like `type PgInt = Int32`.

Comment (3noch) [reply beam-postgres/Database/Beam/Postgres/PgSpecific.hs:1420]: Yeah I'm not really a fan either. It's just that there's not a very mechanical way to verify these kinds of choices strewn throughout the codebase. Perhaps that's just par for the course.

**Pull request merged**